api/include/opentelemetry/std/span.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020, OpenTelemetry Authors
```

Comment:
Abseil's span does not strive to be fully `std::span` compatible, whereas `gsl::span` is making that commitment. This is becoming irrelevant with C++20, where `std::span` is available natively.

Reply:
GSL can be used on any platform. But only if necessary - only if the runtime does not natively provide `std::span`, e.g. it is not needed at all on latest Ubuntu 20.04 (because it already has C++20 compiler with `std::span` ). Also it is not needed on _latest_ Mac OS X. In fact, it is also not needed in Visual Studio 2019 either. I'm thinking here how to add flexibility, as we could have used either `gsl::span` or `absl::span`. I picked GSL because it sets the goal to be fully compatible with the latest C++20 standard.

Reply:
I added a table that lists where `gsl::span` is necessary. Also added paragraph from Abseil `absl::Span` that explains that Abseil, for example, is not targeting to be the substitute for `std::span`, whereas `gsl::span` is a 100% compatible substitute for `std::span` ... Thus, in environments where it's missing - we are using that.